### PR TITLE
KirshApps's First Goodie

### PR DESCRIPTION
**Pull request by kirshapps**
What does your Instant Answer do?

It tells people that I am awesome, who would otherwise not know.

What problem does your Instant Answer solve (Why is it better than organic links)?

It tells the world that I am awesome.

What are some example queries that trigger this Instant Answer?

Example: "duckduckhack kirshapps"

Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)

This Instant Answer is good for everyone.

Where did you hear about DuckDuckHack? (For first time contributors)

By clicking a link from the DuckDuckGo Search Engine.

What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)

image

## Checklist
Please place an 'X' where appropriate.

```
- [X] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
- [X] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
- [X] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
- [X] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
- [X] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        - [X] Google Chrome
        - [X] Firefox
        - [] Opera
        - [] IE 10

    - Windows 7
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 8
        - [] IE 9
        - [] IE 10

    - Windows XP
        - [] IE 7
        - [] IE 8

    - Mac OSX
        - [X] Google Chrome
        - [X] Firefox
        - [] Opera
        - [] Safari

    - iOS (iPhone)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - iOS (iPad)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - Android
        - [] Firefox
        - [] Native Browser
        - [] Google Chrome
        - [] Opera

### CODE REVIEW

File: lib/DDG/Goodie/IsAwesome/kirshapps.pm [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+name "IsAwesome kirshapps";
+description "Tells everyone that kirshapps is awesome";
+primary_example_queries "duckduckhack kirshapps";
+secondary_example_queries "optional -- demonstrate any additional triggers";
+ category "special";
+topics "special_interest", "geek";
+code_url "https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/lib/DDG/Goodie/IsAwesome/kirshapps.pm";
+attribution github => ["https://github.com/kirshapps", "kirshapps"],
+            ;
```
Comment (mintsoft) [inline lib/DDG/Goodie/IsAwesome/kirshapps.pm]: these two lines should be one line like so:
```perl
attribution github => ["https://github.com/kirshapps", "kirshapps"];
```

File: lib/DDG/Goodie/IsAwesome/kirshapps.pm [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+use DDG::Goodie;
+
+zci answer_type => "is_awesome_kirshapps";
+zci is_cached   => 1;
+
+name "IsAwesome kirshapps";
+description "Tells everyone that kirshapps is awesome";
+primary_example_queries "duckduckhack kirshapps";
+secondary_example_queries "optional -- demonstrate any additional triggers";
```
Comment (mintsoft) [inline lib/DDG/Goodie/IsAwesome/kirshapps.pm]: This seconday_example_queries line can be removed

File: lib/DDG/Goodie/IsAwesome/kirshapps.pm [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ category "special";
+topics "special_interest", "geek";
+code_url "https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/lib/DDG/Goodie/IsAwesome/kirshapps.pm";
+attribution github => ["https://github.com/kirshapps", "kirshapps"],
+            ;
+
+triggers start => "duckduckhack kirshapps";
+
+handle remainder => sub {
+	return "kirshapps is awesome and has successfully completed the DuckDuckHack goodie tutorial!"
```
Comment (mintsoft) [inline lib/DDG/Goodie/IsAwesome/kirshapps.pm]: Can you switch this hard-tab out for 4 spaces

File: t/IsAwesome/kirshapps.t [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+use DDG::Test::Goodie;
+
+zci answer_type => "is_awesome_kirshapps";
+zci is_cached   => 1;
+
+ddg_goodie_test(
+    [qw(
+        DDG::Goodie::IsAwesome::kirshapps
+    )],
+    'duckduckhack kirshapps' => test_zci('kirshapps is awesome and has successfully completed the DuckDuckHack Goodie tutorial!'),
```
Comment (javathunderman) [inline t/IsAwesome/kirshapps.t]: Can you decapitalize the g in goodie? This is what is making the tests fail.

Comment (MrChrisW) [inline t/IsAwesome/kirshapps.t]: @javathunderman Probably want to capitilise the "Goodie" in https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1015/files#diff-6afa7237557aa8ba50464e27ce0e8155R16

Comment (javathunderman) [inline t/IsAwesome/kirshapps.t]: Either makes sense...It's up to @kirshapps

Comment (MrChrisW) [inline t/IsAwesome/kirshapps.t]: @javathunderman Hate to be that guy..... :grimacing: As "Goodie" is a *proper noun* it does start with a capital letter and is included in the [tutorial](https://duck.co/duckduckhack/goodie_quickstart#creating-your-first-goodie) this way, but overall it's a tiny change and either way would work.. :smile:

Comment (javathunderman) [inline t/IsAwesome/kirshapps.t]: @MrChrisW You're right, and don't worry...More than usual I am that guy. :smile:

**Comment by kirshapps** [comment]
Thanks @moollaza, I think I've got it now!

File: lib/DDG/Goodie/IsAwesome/kirshapps.pm [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+description "Tells everyone that kirshapps is awesome";
+primary_example_queries "duckduckhack kirshapps";
+category "special";
+topics "special_interest", "geek";
+code_url "https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/lib/DDG/Goodie/IsAwesome/kirshapps.pm";
+attribution github => ["https://github.com/kirshapps", "kirshapps"];
+triggers start => "duckduckhack kirshapps";
+handle remainder => sub {
+    return if $_;
+	return "kirshapps is awesome and has successfully completed the DuckDuckHack goodie tutorial!"
```
Comment (mintsoft) [inline lib/DDG/Goodie/IsAwesome/kirshapps.pm]: @kirshapps Can you fix this hard-tab here? It should be 4 spaces :+1:

Comment (kirshapps) [inline lib/DDG/Goodie/IsAwesome/kirshapps.pm]: Hi @mintsoft. How exactly would I do that?

Comment (mintsoft) [inline lib/DDG/Goodie/IsAwesome/kirshapps.pm]: @kirshapps sorry for the slow reply; you could just delete the tab character and manually type 4 spaces; however the Codio editor should already be configured for spaces/"soft tabs":
image

Comment (kirshapps) [inline lib/DDG/Goodie/IsAwesome/kirshapps.pm]: Thanks @mintsoft.

**Comment by mintsoft** [comment]
Cool thanks @kirshapps Congrats on your first Goodie!

**Comment by moollaza** [comment]
@kirshapps congrats! It's live: https://duckduckgo.com/?q=duckduckhack+kirshapps&ia=answer

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's & Issues or even submit more Instant Answers!

**Pull request merged**